### ISSUE TITLE
fit_generator with variable samples size

### ISSUE THREAD

**Issue by Wissben** [keras-team/keras#12848]
Hello dear ML scientist !
I'd like to ask you a question, right now i'm trying to develop a model to solve a slot_filling problem using a seq2seq architecture implemented in Keras.
It happens that I need to use the `model.fit_generator` method to train on my data, and my dataset is formed of a bunch of Xs and Ys of the form :
X_i = w1 w2 w3 ... wn, a sequence of words forming a sentence.
Y_i = t1 t2 t3 ... tn , a bunch  of tags associated to each word of X_i
since each pair of X_i,Y_i can be of different length, and i don't want to use padding, i regrouped all the Xs,Ys of same length in a dictionary of the form
`{length1 : [(x,y) , (x',y') ... ], lentgth2 : [(x'',y''),(x''',y''') ...] ...}`. Now my question is how do write a proper generator to yield all the samples of same length in series of batches of same size knowing that the argument `steps_per_epoch` of the `model.fit_generator` is supposed to be equals to num_samples/batch_size and that my num_samples varies from one group of same length samples to another.
PS : if it's not the right channel to ask just tell me
Thanks
